assets/js/ajax-core.js

```diff
@@ -129,22 +129,43 @@ jQuery(function($){
 
                 ev.preventDefault();
 
-                var form	=	$(el),
-                    buttons = 	form.find(':submit');
-
-                if( form.data('_cf_manual') ){
-                    form.find('[name="cfajax"]').remove();
+                var $form	=	$(el),
+                    buttons = 	$form.find(':submit');
+                var pending = [];
+
+				/**
+                 * This event is triggered directly before the request for form submission is made
+                 *
+                 * Runs after cf.form.submit
+                 *
+                 * @since 1.8.0
+				 */
+				$( document ).trigger( 'cf.form.request', {
+                    $form: $form,
+                    formIdAttr: $form.attr( 'id' )
+                });
+
+				//Check if any cf2 fields are blocking submit
+				var cf2 = window.cf2[ $form.attr( 'id' ) ];
```

Comment:
@New0 I shoved it into a window scoped object I was already using. That's better than a window scope object, I think.